### ISSUE TITLE
Export WSL distro to stdout ( wsl --export <distro> - ) doesn't work

### ISSUE THREAD

**Issue by okibcn** [microsoft/WSL#9058]
### Version

Microsoft Windows [Version 10.0.25227.1010]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.68.1

### Distro Version

Arch custom image.

### Other Software

WSL version: 0.70.4.0
Kernel version: 5.15.68.1
WSLg version: 1.0.45
MSRDC version: 1.2.3575
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.25227.1010

### Repro Steps

Any of the following options should work, and they doesn't:

1)
```powershell
wsl --export Arch - | Set-Content "./dump.tar"
```
2)
```powershell
wsl --export Arch - | 7z a -si -mx9 "./dump.tar.7z"
```
3)
```powershell
wsl --export distro - | wsl --import newdistro "path/to/newdistro" -
```

### Expected Behavior

1) a tar file with the ump of the distro.

2) a 7z file with a tar inside containing a dump of the distro filesystem.

3) a duplicated distro.

### Actual Behavior

1) a tar file with garbage content.

2) a 7z file with a tar inside containing garbage.

3) the process remains forever without doing anything.

**It seems the stdout option of the --export option doesn't work as expected, not even when piped directly into the wsl --import from pipe.**

**Comment by okibcn** [comment]
Please note that ```wsl --export Arch "./dump.tar"``` works. The problem is limited to the stdout feature.

**Comment by okibcn** [comment]
The feature works, but only in a cnd line. For instance, The way to duplicate a distro doesn't work in powershell by doing:
```powershell
wsl --export distro - | wsl --import newdistro "path/to/newdistro" -
```
However, it works when invoking the command interpreter:
```powershell
cmd /c "wsl --export distro - | wsl --import newdistro ""path/to/newdistro"" -"
```
If the pipe used in wsl export doesn't work in powershell, it should have a warning or something in the help for the export and import functions.

**Comment by benhillis** [comment]
Interesting, I was unaware this didn't work in powershell, I will investigate.

**Comment by OneBlue** [comment]
Thank you for reporting this @okibcn.

Unfortunately, this is caused by [how powershell handles pipes](https://brianreiter.org/2010/01/29/powershells-object-pipeline-corrupts-piped-binary-data/)

TLDR: Powershell pipes expect strings and don't handle binary data correctly.

Sadly there isn't any easy way for wsl to detect that it's called from inside powershell so outputting a warning isn't an option either.

**Comment by okibcn** [comment]
@OneBlue I wasn't suggesting any detection routine, but a note to show the limitations. At this time that is misleading, as the feature is useless in PowerShell.  Maybe a proper solution would be adding that information in the help text, explaining that using standard output during export and standard input for the import functions only works properly in cmd and not in pwsh.
